### ISSUE TITLE
How to set different learning rate for different layers?

### ISSUE THREAD

**Issue by yushuinanrong** [vlfeat/matconvnet#431]
Hi guys,

I'm new to the matconvnet. I want to fine-tune a pre-trained model and set large learning rate to the last layers which I add myself, and small learning rate to the original layers which I've not touched. I'm not sure how to do this in matconvnet? Any idea will be appreciated!

Thanks,
Jiawei

**Comment by yushuinanrong** [comment]
I think there is no clear way to assign different weights to different layers, which is quite inconvenient compared to caffe. I hope I'm wrong here.

**Comment by fengyunxiaozi** [comment]
hi, it's easy to do this. for each layer, you can set "net.layers{i}.learningRate = [a,b]“. see the example. By the way, there are many similar issues that proposed by others, you can search them.

**Comment by yushuinanrong** [comment]
@fengyunxiaozi 
Thanks for your answer!
Though, I'm still not 100% understand. I attach a portion of my code below. Where I define learning rate for the new layer. However, I find in cnn_train.m file opt.learningrate (default value is 0.001) is a global variable, which I think can overwrite my predefined learningrate. Could you explain more?

Thanks.
 
% Transfer learning / adapt models.
net = load('imagenet-caffe-alex.mat');
net.layers = net.layers(1:end-2);
net.layers{end+1} = struct('type', 'conv', ...
'weights', {{0.005*randn(1,1,4096,145, 'single'), zeros(1,145,'single')}}, ...
'learningRate', [0.05 0.0], ...
'stride', [1 1], ...
'pad', [0 0 0 0]) ;
net.layers{end+1} = struct('type', 'softmaxloss') ;

**Comment by fengyunxiaozi** [comment]
Here 'learningRate', [0.05 0.0] is used to set the layer learning rate.  There is no need to change cnn_ttrain function. If you want to set the basic learning rate of the network, just change in the layer
initialization function.  See the examples provided in MatConvNet.

**Comment by yushuinanrong** [comment]
@fengyunxiaozi 
I figure our eventually. Really appreciate your help!
Best

Jiawei
